### PR TITLE
Allow Clipper2 to be supplied externally when using Manifold as a submodule

### PR DESCRIPTION
**Pull request by kintel**
* Checks `Clipper2_FOUND` before attempting to find Clipper2
* Add imported Clipper2 library _only_ when trying pkg_config

### CODE REVIEW

File: cmake/manifoldDeps.cmake [hunk 1]
```diff
@@ -78,26 +78,27 @@ endif()
...
-    )
-    if(WIN32)
-      set_property(
-        TARGET Clipper2
-        PROPERTY IMPORTED_IMPLIB ${Clipper2_LINK_LIBRARIES}
-      )
-    endif()
-    target_include_directories(Clipper2 INTERFACE ${Clipper2_INCLUDE_DIRS})
-  else()
+      if(Clipper2_FOUND)
```
Comment (elalish) [inline cmake/manifoldDeps.cmake:85]: Can this condition ever be hit? Two lines up we say that clipper2 was not found.

Comment (pca006132) [reply cmake/manifoldDeps.cmake:85]: pkg-config can set this to true after the call to `pkg_check_modules`.

Comment (elalish) [reply cmake/manifoldDeps.cmake:85]: Ah, got it, thanks.

**Comment by elalish** [comment]
What are you waiting for to make this not a Draft?

**Comment by kintel** [comment]
> What are you waiting for to make this not a Draft?

My own manual testing across platforms, which takes some time. I mostly wanted to make sure this is conceptually ok with you, plus make sure Manifold's tests pass.

**Comment by kintel** [comment]
This is good to go now!

**Comment by pca006132** [comment]
LGTM. I'm wondering if we should do something similar for other dependencies, e.g. tbb.

**Comment by kintel** [comment]
Yeah, it could make sense to align the deps lookups. I didn't run into the same issue with tbb as we're always supplying it as a system dependency from the OpenSCAD side.

**Comment by kintel** [comment]
Another idea would be to hide all the details in custom Find macros, but I didn't want to touch too much as it's so time consuming to test it..

**Review by pca006132 — APPROVED** [review]

**Review by elalish — APPROVED** [review]
Thanks!

**Pull request merged**